src/kakarot/model.cairo

```diff
@@ -27,6 +45,23 @@ namespace model {
         value: felt,
     }
 
+    // @dev Stores all data relevant to the current execution context
+    // @param call_context - call context data
+    // @param program_counter - keep track of the current position in the program as it is being executed
+    // @param stopped - boolean that state if the current execution is halted
+    // @param return_data - region used to return a value after a call
+    // @param return_data_len - return_data length
+    // @param stack - current execution context stack
+    // @param memory - current execution context memory
+    // @param gas_used - gas consumed by the current state of the execution
+    // @param gas_limit - maximum amount of gas for the execution
+    // @param gas_price - the amount to pay per unit of gas
+    // @param starknet_contract_address - starknet address of the contract interacted with
+    // @param evm_contract_address - evm address of the contract interacted with
+    // @param calling_context - parent context of the current execution context
+    // @param sub_context - child context of the current execution context
```

Comment:
these might not exist, I mean, there are always in the struct but they can be empty

Reply:
I added `(optional)` at the end, not sure if it's the most accurate term

Reply:
you right it's not accurate because there are no optional args in cairo. Better saying smth like "can be empty when context is root context, see `ExecutionContext.is_root`" etc.